sf-stream/src/protos/extractor.proto

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+message Transaction {
+  uint64 timestamp = 1;
+  uint64 version = 2;
+  TransactionInfo info = 3;
+  uint64 epoch = 4;
+  uint64 block_height = 5;
+
+  enum TransactionType {
+    GENESIS = 0;
+    BLOCK_METADATA = 1;
+    STATE_CHECKPOINT = 2;
+    USER = 3;
+  }
+  
+  TransactionType type = 6;
```

Comment:
We could argue here that since we are using `oneof` below, we could probably determine the type just by check at the actual value of the variant of `oneof`. In Rust, `oneof` are mapped to an enum so it's quite ease to match the actual payload's type:

```
match trx.trxn_data {
  BlockMetadata(_) => ...
  User(_) => ...
}
```

There is however a simplicity for user for having the type enum mainly because it's kind of "easier" to see how to determine the type of a transaction, so there is still benefits to have both.

Hard choice to be honest.

Reply:
Agreed in rust this is superflous, but ended up leaving it in for python/ts, where this becomes a series of `if null/else` otherwise; I went back and forth on this choice as well :-)

Reply:
Yeah not easy, I had first in my Proto I quickly drafted then remove it. It does not hurt having both and in doubt, let's keep it :)